packages/core/src/components/Sequence/Sequence.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { Fragment, FC, useState } from 'react';
+import {
+  Stepper,
+  Step,
+  StepContent,
+  StepLabel,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+import SequenceFooter from './SequenceFooter';
+
+const useStyles = makeStyles((theme: any) => ({
+  content: {
+    padding: theme.spacing(0, 4, 4, 4),
+    backgroundColor: 'white',
+  },
+  completionBox: {
+    padding: theme.spacing(3),
+  },
+}));
+
+type Orientation = 'horizontal' | 'vertical';
+
+export type StepActions = {
+  showNext?: boolean;
+  canNext?: () => boolean;
+  onNext?: () => void;
+  nextStep?: (current: number, last: number) => number;
+  nextText?: string;
+
+  showBack?: boolean;
+  backText?: string;
+  onBack?: () => void;
+
+  showRestart?: boolean;
+  canRestart?: () => boolean;
+  onRestart?: () => void;
+  restartText?: string;
+
+  completedButton?: JSX.Element;
+};
+
+export type StepType = {
+  title: string;
+  actions?: StepActions;
+  content: JSX.Element;
+};
+
+export interface SequenceProps {
+  completed?: StepType;
+  elevation?: boolean;
+  orientation: Orientation;
+  onSequenceStepChange?: (prevIndex: number, nextIndex: number) => void;
+  steps: StepType[];
+  footerProps?: any;
+}
+
+const Sequence: FC<SequenceProps> = ({
+  completed,
+  elevation,
+  orientation,
+  onSequenceStepChange,
+  steps,
+  footerProps,
+}) => {
+  const classes = useStyles();
+  const [stepIndex, setStepIndex] = useState<number>(0);
+  const [stepArray, setStepArray] = useState<number[]>([0]);
+  const isVertical = orientation === 'vertical';
+  const horizontalStep =
+    !isVertical && steps[stepIndex] ? steps[stepIndex].content : false;
+
+  return (
+    <Fragment>
+      <Stepper
+        activeStep={stepIndex}
+        orientation={orientation}
+        elevation={elevation ? 2 : 0}
+      >
+        {steps.map((step, i) => (
+          <Step key={`step${i}`}>
```

Comment:
I think key could just be `i` here right?

Reply:
should probably be some id actually

Reply:
but will change it to use the index for now